src/imap.rs

```diff
@@ -952,15 +966,35 @@ impl Session {
                     return Ok(());
                 }
                 Err(err) => {
+                    if context.should_delete_to_trash().await? {
```

Comment:
If the server doesn't support MOVE we shouldn't get here at all, `self.can_move()` above is for that. Of course if we detect it properly. So, i suppose here that it can be a temporary error and after a retry messages could be successfully moved. But if we instead copy-delete them, they will be archived for Gmail f.e. I'd leave it as is, because if MOVE fails for Gmail for some reason, it is still unlikely that a copy-delete will succeed